pkgs/tools/misc/kb/default.nix

```
@@ -0,0 +1,51 @@
+{ lib
+, fetchFromGitHub
+, python3Packages
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "kb";
+  version = "0.1.7";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "gnebbia";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0hjvf1ycwf4f8rnrshj30ij56mkji2ic3ibpschyvng5jslh1h9b";
+  };
+
+  preBuild = ''
+    substituteInPlace setup.py \
+      --replace \
+        "install_requires=[\"colored\",\"toml\",\"attr\",\"attrs\",\"gitpython\"]," \
+        "install_requires=[\"colored\",\"toml\",\"attrs\",\"gitpython\"],"
+  '';
+
+  propagatedBuildInputs = with python3Packages; [
+    colored
+    toml
+    attrs
+    gitpython
+  ];
+
+  nativeCheckInputs = [ python3Packages.pytest ];
```

This is the preferred way to run pytest, but I ran into this error
```suggestion
  nativeCheckInputs = [ python3Packages.pytestCheckHook ];
```
```console
ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: --cov=kb --cov-report
  inifile: /build/source/setup.cfg
  rootdir: /build/source
```

---

We might need to patch out the `--cov` flags for this to work

---

We usually patch out the cov flags because they are not useful for us.

---

> This is the preferred way to run pytest, but I ran into this error
> 
> ```
> ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
> pytest: error: unrecognized arguments: --cov=kb --cov-report
>   inifile: /build/source/setup.cfg
>   rootdir: /build/source
> ```

same here.

```
ERROR: usage: __main__.py [options] [file_or_dir] [file_or_dir] [...]
__main__.py: error: unrecognized arguments: --cov=kb --cov-report
  inifile: /build/source/setup.cfg
  rootdir: /build/source
```

btw, how do I find documentation for these options?  I always read other people packages and try to make it work by myself.

---

I believe this is part of pytest-cov, but there is no reason for us to do coverage reporting